### ISSUE TITLE
identify constraint violations in the proposed plan

### ISSUE THREAD

**Issue by rdicosmo** [aeolus-project/metis#1]
It would be extremely useful to read all the specifications of the Aeolus model (including conflicts and capacity constraints), and after finding a plan, verify which constraints are violated, and emit an appropriate warning (maybe different classes of warning for the 3 different violations... provides abused, requirements undersatisfied, and conflicts violated).

This will help the used decide whether the plan is viable, or allow an iterative approach to find a better plan when needed.

**Comment by talascu** [comment]
Indeed. As odd as it may sound, I never thought about this possibility.

At the moment we are focusing on the following development directions:
1. conflict detection. If conflicts are "global" we could issue a warning when a potentially conflicting state is entered.
2. conflict avoidance. If conflicts are "local" we could generate the plan, then from it generate a "conflict-graph" and finally find a minimal set of machines/locations that need to be used in order to avoid conflicts.

As for capacity constraints we are starting to think about the possibilities to integrate them, but as for now we don't have much material on this topic.

If the above issues turn out to be too complex, we could, in a first phase, do what you suggest and report violations of the deployment plan.

**Comment by rdicosmo** [comment]
On Tue, Dec 03, 2013 at 02:05:47AM -0800, Tudor A. Lascu wrote:

> Indeed. As odd as it may sound, I never thought about this possibility.
> 
> At the moment we are focusing on the following development directions:
> 1. conflict detection. If conflicts are "global" we could issue a warning when
> a potentially conflicting state is entered.
> 2. conflict avoidance. If conflicts are "local" we could generate the plan,
> then from it generate a "conflict-graph" and finally find a minimal set of
> machines/locations that need to be used in order to avoid conflicts.

OK!

> As for capacity constraints we are starting to think about the possibilities to
> integrate them, but as for now we don't have much material on this topic.
> 
> If the above issues turn out to be too complex, we could, in a first phase, do
> what you suggest and report violations of the deployment plan.

Looking forward to see the results :-)

> —
> Reply to this email directly or view it on GitHub.*

## 

Roberto Di Cosmo

---

Professeur               En delegation a l'INRIA
PPS                      E-mail: roberto@dicosmo.org
Universite Paris Diderot WWW  : http://www.dicosmo.org
Case 7014                Tel  : ++33-(0)1-57 27 92 20
5, Rue Thomas Mann  
F-75205 Paris Cedex 13   Identica: http://identi.ca/rdicosmo

## FRANCE.                  Twitter: http://twitter.com/rdicosmo

Attachments:
MIME accepted, Word deprecated

##       http://www.gnu.org/philosophy/no-word-attachments.html

Office location:

Bureau 3020 (3rd floor)
Batiment Sophie Germain
Avenue de France

## Metro Bibliotheque Francois Mitterrand, ligne 14/RER C

GPG fingerprint 2931 20CE 3A5A 5390 98EC 8BFC FCCA C3BE 39CB 12D3
